include/fiction/networks/views/input_ordering_view.hpp

```diff
@@ -0,0 +1,525 @@
+//
+// Created by benjamin on 11.04.23.
+//
+
+#ifndef FICTION_INPUT_ORDERING_VIEW_HPP
+#define FICTION_INPUT_ORDERING_VIEW_HPP
+
+#include "fiction/traits.hpp"
+#include "fiction/utils/network_utils.hpp"
+
+#include <mockturtle/networks/detail/foreach.hpp>
+#include <mockturtle/traits.hpp>
+#include <mockturtle/views/fanout_view.hpp>
+#include <mockturtle/views/immutable_view.hpp>
+
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <iostream>
+#include <optional>
+#include <vector>
+
+namespace fiction
+{
+
+/**
+ * A variation of mockturtle::topo_view that computes the topological order after sorting the inputs according to the
+ * connections of their fanouts. This is part of the Distribution Newtork I: Ordering_Network
```

Comment:
Maybe you can give a bit more of a description of how PIs are sorted exactly and what the benefits are. Imagine a user who thinks about applying this algorithm. What would they need to know?